### PR TITLE
feat: parallelize span, trace total and event fetches

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Improve performance of trace and span queries by fetching the result rows and total in parallel.

### Testing
Reverified unit, e2e tests

### CODE REVIEW

File: gateway-service-impl/src/test/resources/configs/gateway-service/application.conf [hunk 1]
```diff
@@ -17,6 +17,9 @@ attributes.service.config = {
   host = localhost
   port = 9012
 }
+query.executor.config = {
```
Comment (kotharironak) [inline gateway-service-impl/src/test/resources/configs/gateway-service/application.conf:20]: Can we add the helm too?

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java [hunk 2]
```diff
@@ -79,6 +82,8 @@ public GatewayServiceImpl(Config appConfig) {
     QueryServiceClient queryServiceClient =
         new QueryServiceClient(new QueryServiceConfig(qsConfig));
     int qsRequestTimeout = getRequestTimeoutMillis(qsConfig);
+    ExecutorService queryExecutor =
+        QueryExecutorServiceFactory.buildExecutorService(QueryExecutorConfig.from(appConfig));
```
Comment (kotharironak) [inline gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java:86]: are we use `ServiceFactory` suffix other places? if, not, `QueryExecutorServiceFactory` -> `QueryExecutorFactory` ?

File: gateway-service-impl/src/test/resources/configs/gateway-service/application.conf (hunk 1 above)
Comment (kotharironak) [reply gateway-service-impl/src/test/resources/configs/gateway-service/application.conf:20]: can be added in follow-up.

**Review by kotharironak — APPROVED** [review]
lgtm

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java (hunk 2 above)
Comment (aaron-steinfeld) [reply gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java:86]: So I went back and forth on this - agreed we should be consistent, but the actual returned interface is an ExecutorService, so it's a factory for that, and the role of that executor service is querying.

Will proceed, but let's figure out the naming and can go back and make consistent.

File: gateway-service-impl/src/test/resources/configs/gateway-service/application.conf (hunk 1 above)
Comment (aaron-steinfeld) [reply gateway-service-impl/src/test/resources/configs/gateway-service/application.conf:20]: the helm config map template takes in a blob for application conf, so no changes are needed there (I could have overridden this default, but that didn't seem necessary):

https://github.com/hypertrace/gateway-service/blob/main/helm/values.yaml#L67-L76

**Pull request merged**